PlatformLevelTechempower/AsciiString.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Text;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Infrastructure;
+using Microsoft.AspNetCore.Server.Kestrel.Internal.System;
+
+namespace PlatformLevelTechempower
+{
+    public struct AsciiString : IEquatable<AsciiString>
+    {
+        private readonly byte[] _data;
+
+        public AsciiString(byte[] bytes) => _data = bytes;
+
+        public AsciiString(string s) => _data = Encoding.ASCII.GetBytes(s);
+
+        public int Length => _data.Length;
+
+        public static implicit operator ReadOnlySpan<byte>(AsciiString str) => str._data;
+        public static implicit operator byte[] (AsciiString str) => str._data;
+
+        public static implicit operator AsciiString(byte[] bytes) => new AsciiString(bytes);
+        public static implicit operator AsciiString(string str) => new AsciiString(str);
+
+        public override string ToString() => HttpUtilities.GetAsciiStringNonNullCharacters(_data);
+        public static explicit operator string(AsciiString str) => str.ToString();
+
+        public bool Equals(AsciiString other) => ReferenceEquals(_data, other._data) || SequenceEqual(_data, other._data);
+        private bool SequenceEqual(byte[] data1, byte[] data2) => new Span<byte>(data1).SequenceEqual(data2);
+
+        public static bool operator ==(AsciiString a, AsciiString b) => a.Equals(b);
+        public static bool operator !=(AsciiString a, AsciiString b) => !a.Equals(b);
+        public override bool Equals(object other) => (other is AsciiString) && Equals((AsciiString)other);
+
+        public override int GetHashCode()
+        {
+            var data = _data;
```

Comment:
Where did you get this impl?

Reply:
Copied `string.GetLegacyNonRandomizedHashCode()` dumped all the #if s

https://github.com/dotnet/coreclr/blob/c90a41bb6b9b5c5cc7f8f29a9d06aa2f9d0e51fd/src/mscorlib/src/System/String.Comparison.cs#L982-L1036

Reply:
Added comment